### PR TITLE
Deployed list CRUD functionality

### PR DESCRIPTION
**Pull request by demesameneshoa**
## In this PR I executed
- [x] Removing all hardcoded items from the tasks array.
- [x] Creating a new JavaScript file for the new functionality.
- [x] Implementing a function for adding a new task (add a new element to the array).
- [x] Implementing a function for deleting a task (remove an element from the array).
- [x] Implementing a function for editing task descriptions.

### CODE REVIEW

File: src/modules/handletodo.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+  const newindex = Number(delteIndex);
+  toDoLists.splice(newindex - 1, 1);
+  // reset indexes of all array values
+  const maxIndex = toDoLists.length;
+  for (let i = 0; i < maxIndex; i += 1) {
+    toDoLists[i].taskindex = i;
+  }
+  addTostorage();
+  window.location.reload();
+};
```
Comment (Shisui6) [inline src/modules/handletodo.js]: According to the project requirements, deleting a task should update all remaining items' indexes, so they represent the current list order and are unique(i.e. if you're deleting the first task index 1 from the list, the index of the next task(2) should set to 1). This functionality has not been correctly implemented as the indexes of the remaining tasks start from zero as seen in the video below:


https://github.com/demesameneshoa/to-do-list/assets/37116322/b80279f8-ced6-44ff-bca4-183a1f5b7e23



Kindly update your app so the delete functionality is implemented correctly🙏

File: src/index.js [hunk 2]
```diff
@@ -1,7 +1,55 @@
...
+  editTask,
+  removeTask,
+} from './modules/handletodo.js';
 
-// Adding main component to the HTML file
-const mainContainer = document.querySelector('.main');
-mainContainer.appendChild(render());
+render();
+// add task when text is submitted
+const taskinput = document.querySelector('.task-desc');
```
Comment (Shisui6) [inline src/index.js:12]: [Optional] Currently, your app refreshes every time a task is added and removed from the list. To improve the user experience, I suggest you prevent this refresh action.

**Review by Shisui6 — CHANGES_REQUESTED** [review]
Hi @demesameneshoa, 

# Requires Changes 🍀
You have done a great job so far!👏👏
You still need to work on a few issues before moving on to the next project, but you're almost there🪲!

## Highlights🎯
- CRUD functions implemented ✅


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions 🪶
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._

You can also consider the following:

N/A

Cheers and Happy coding!🍻💻

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid, unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561), you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/index.js (hunk 2 above)
Comment (Qoosim) [inline src/index.js:12]: - [OPTIONAL] As suggested by the previous reviewer, kindly ensure that your page does not refresh at every operation you perform as to give the best user experience.

**Review by Qoosim — APPROVED** [review]
#  *Status: Approved* :tada: :heavy_check_mark: 

Hi @demesameneshoa, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
-  No linter errors. :heavy_check_mark: 
-  Good README file. :heavy_check_mark: 
-  Correct GitHub Flow. :heavy_check_mark: 



## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**